### PR TITLE
chore(options): Remove unused options

### PR DESCRIPTION
**Pull request by jan-auer**
These options are no longer used in code and have all been removed from options
automator. To make the list of available options more manageable, we remove the
option from code.

### CODE REVIEW

File: tests/sentry/sentry_metrics/test_gen_metrics_multiprocess_steps.py [hunk 1]
```diff
@@ -408,12 +408,7 @@ def test_process_messages_default_card_rollout(set_sentry_option: Callable[...,
 
     outer_message = Message(Value(message_batch, last.committable))
 
-    with set_sentry_option(
-        "sentry-metrics.cardinality-limiter.orgs-rollout-rate",
-        1.0,
```
Comment (jan-auer) [inline tests/sentry/sentry_metrics/test_gen_metrics_multiprocess_steps.py:413]: 1.0 was the default, but it was also not used.

**Review by Dav1dde — APPROVED** [review]

File: tests/sentry/sentry_metrics/test_gen_metrics_multiprocess_steps.py (hunk 1 above)
Comment (ayirr7) [reply tests/sentry/sentry_metrics/test_gen_metrics_multiprocess_steps.py:413]: IIRC there was a test that used this option? Should we modify that test accordingly?

Comment (jan-auer) [reply tests/sentry/sentry_metrics/test_gen_metrics_multiprocess_steps.py:413]: This is the only test and it set this option to the default. I suppose this change is all we need?

File: src/sentry/options/defaults.py [hunk 2]
```diff
@@ -1292,27 +1280,11 @@
...
-)
-
 # Option to enable orjson for JSON parsing in reconstruct_messages function
 register(
     "sentry-metrics.indexer.reconstruct.enable-orjson", default=0.0, flags=FLAG_AUTOMATOR_MODIFIABLE
 )
 
-# Option to enable direct storage queries for meta queries in the metrics layer
-register(
-    "sentry-metrics.metrics-layer.use-storage-direct-meta-queries",
```
Comment (evanh) [inline src/sentry/options/defaults.py:1311]: @enochtangg can this be removed?

Comment (enochtangg) [reply src/sentry/options/defaults.py:1311]: 👍

**Review by enochtangg — APPROVED** [review]

**Pull request merged**